### PR TITLE
refactor(accordion): move to the new architecture

### PR DESCRIPTION
**Pull request by aesteves60**
⚠ BREAKING CHANGES ⚠

   - `OsdsAccordion` is no longer part of `@ovhcloud/ods-stencil` package
   - `OsdsAccordion` is now accessible by adding either `ods-component-accordion` or `ods-components` dependencies
   - `@ovhcloud/ods-stencil-accordion` lib has been replaced by `@ovhcloud/ods-component-accordion`
   - `@ovhcloud/ods-stencil-accordion-react` lib has been replaced by `@ovhcloud/ods-component-accordion-react`
  -  `@ovhcloud/ods-stencil-accordion-vue` lib has been replaced by `@ovhcloud/ods-component-accordion-vue`
    
  - `OdsAccordionSize` has been removed from `@ovhcloud/ods-core` and is replaced by `ODS_ACCORDION_SIZE` from `@ovhcloud/ods-component-accordion`
  -  `OdsAccordionSizeList` has been removed from `@ovhcloud/ods-core` and is replaced by `ODS_ACCORDION_SIZES` from `@ovhcloud/ods-component-accordion`
  -  `OdsAccordionSizeUnion` has been removed from `@ovhcloud/ods-core`
    
   - `OsdsAccordionGroup` is no longer part of `@ovhcloud/ods-stencil package`
   - `OsdsAccordionGroup` is now accessible by adding either `ods-component-accordion` or `ods-components` dependencies
    
    
    
   - `@ovhcloud/ods-component-accordion` exports only `OdsAccordionGroupMethod`, `OdsAccordionAttribute`, `OdsAccordionEvents`, `OdsAccordionToggleEvent` as component interface.

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-storybook#1123.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#1129.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#1129.1 ✘  
* 
  * build ✘

File: packages-new/components/accordion/src/components/osds-accordion-group/osds-accordion-group.tsx [hunk 1]
```diff
@@ -1,26 +1,16 @@
 import { Component, Element, Host, Method, h } from "@stencil/core";
-import {
-  OdsAccordion,
-  OdsAccordionGroup,
-  OdsAccordionGroupController,
-  OdsAccordionGroupEvents,
-  OdsAccordionGroupMethods,
-} from "@ovhcloud/ods-core";
-import { OdsStencilEvents, OdsStencilMethods } from "@ovhcloud/ods-stencil/libraries/stencil-core";
 import { HTMLStencilElement } from "@stencil/core/internal";
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion-group/osds-accordion-group.tsx:2]: type

File: packages-new/components/accordion/src/components/osds-accordion-group/osds-accordion-group.tsx [hunk 2]
```diff
@@ -1,26 +1,16 @@
...
+import { OsdsAccordion } from "../osds-accordion/osds-accordion";
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion-group/osds-accordion-group.tsx:3]: type

File: packages-new/components/accordion/src/components/osds-accordion-group/osds-accordion-group.tsx [hunk 3]
```diff
@@ -1,26 +1,16 @@
...
+import { OdsAccordionGroupController } from "./core/controller";
+import { OdsAccordionGroupMethods } from "./interfaces/methods";
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion-group/osds-accordion-group.tsx:5]: type

File: packages-new/components/accordion/src/components/osds-accordion/constants/default-attributes.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { OdsAccordionAttributes } from '../interfaces/attributes';
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/constants/default-attributes.ts:1]: type

File: packages-new/components/accordion/src/components/osds-accordion/constants/default-attributes.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+import { ODS_ACCORDION_SIZE } from './accordion-size';
+import { OdsThemeColorIntent } from '@ovhcloud/ods-theming';
+
+const DEFAULT_ATTRIBUTE: OdsAccordionAttributes = Object.freeze({
+  color: OdsThemeColorIntent.info,
+  contrasted: false,
+  size: ODS_ACCORDION_SIZE.md,
+  opened: false,
+  disabled: false
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/constants/default-attributes.ts:10]: alphabetical order would be nice

File: packages-new/components/accordion/src/components/osds-accordion/core/controller.spec.ts [hunk 6]
```diff
@@ -1,28 +1,28 @@
 import { EventEmitter } from '@stencil/core';
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/core/controller.spec.ts:1]: type

File: packages-new/components/accordion/src/components/osds-accordion/core/controller.spec.ts [hunk 7]
```diff
@@ -1,28 +1,28 @@
...
-import { OdsAccordion } from './ods-accordion';
-import { OdsAccordionController } from './ods-accordion-controller';
-import { OdsAccordionGroup } from '../accordion-group/ods-accordion-group';
-import { OdsAccordionGroupMock } from './../accordion-group/ods-accordion-group-mock';
-import { OdsAccordionMock } from './ods-accordion-mock';
+import { HTMLStencilElement } from '@stencil/core/internal';
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/core/controller.spec.ts:2]: type

File: packages-new/components/accordion/src/components/osds-accordion/core/controller.ts [hunk 8]
```diff
@@ -1,15 +1,8 @@
-import { OdsAccordion } from './ods-accordion';
-import { OdsComponentController } from '../ods-component-controller';
+import { OsdsAccordion } from "../osds-accordion";
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/core/controller.ts:1]: type

File: packages-new/components/accordion/src/components/osds-accordion/core/controller.ts [hunk 9]
```diff
@@ -1,15 +1,8 @@
...
 
-/**
- * common controller logic for accordion component used by the different implementations.
- * it contains all the glue between framework implementation and the third party service.
- */
-export class OdsAccordionController extends OdsComponentController<OdsAccordion> {
+export class OdsAccordionController {
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/core/controller.ts:3]: ideally put export on bottom

File: packages-new/components/accordion/src/components/osds-accordion/interfaces/attributes.ts [hunk 10]
```diff
@@ -1,16 +1,19 @@
-import { OdsAccordionSize } from './ods-accordion-size';
-import { OdsComponentAttributes } from '../ods-component-attributes';
+import { ODS_ACCORDION_SIZE } from '../constants/accordion-size';
 import { OdsThemeColorIntent } from '@ovhcloud/ods-theming';
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/interfaces/attributes.ts:2]: type for both

File: packages-new/components/accordion/src/components/osds-accordion/interfaces/attributes.ts [hunk 11]
```diff
@@ -1,16 +1,19 @@
...
   color?: OdsThemeColorIntent;
   /** contrasted or not: see component principles */
   contrasted?: boolean;
   /** size: see component principles */
-  size?: OdsAccordionSize;
+  size?: ODS_ACCORDION_SIZE;
   /** opened or not */
   opened?: boolean;
   /** disabled: see component principles */
   disabled?: boolean;
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/interfaces/attributes.ts:14]: alpha order?

File: packages-new/components/accordion/src/components/osds-accordion/interfaces/attributes.ts [hunk 12]
```diff
@@ -1,16 +1,19 @@
-import { OdsAccordionSize } from './ods-accordion-size';
-import { OdsComponentAttributes } from '../ods-component-attributes';
+import { ODS_ACCORDION_SIZE } from '../constants/accordion-size';
 import { OdsThemeColorIntent } from '@ovhcloud/ods-theming';
 
-export interface OdsAccordionAttributes extends OdsComponentAttributes {
+interface OdsAccordionAttributes {
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/interfaces/attributes.ts:4]: same for plural/singular

File: packages-new/components/accordion/src/components/osds-accordion/interfaces/events.ts [hunk 13]
```diff
@@ -0,0 +1,12 @@
+import { EventEmitter } from '@stencil/core';
+
+interface OdsAccordionEvents {
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/interfaces/events.ts:3]: same for plural/singular

File: packages-new/components/accordion/src/components/osds-accordion/interfaces/events.ts [hunk 14]
```diff
@@ -0,0 +1,12 @@
+import { EventEmitter } from '@stencil/core';
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/interfaces/events.ts:1]: type

File: packages-new/components/accordion/src/components/osds-accordion/interfaces/type.ts [hunk 15]
```diff
@@ -0,0 +1,7 @@
+import { OdsAccordionEvents } from "./events";
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/interfaces/type.ts:1]: you can put this directly in the `events.ts` file

File: packages-new/components/accordion/src/components/osds-accordion/osds-accordion.e2e.screenshot.ts [hunk 16]
```diff
@@ -1,16 +1,18 @@
 import { E2EElement, E2EPage, newE2EPage } from "@stencil/core/testing";
-import { OdsAccordionAttributes, OdsAccordionSizeList, OdsComponentAttributes2StringAttributes, odsAccordionDefaultAttributes } from "@ovhcloud/ods-core";
-import { OdsCreateAttributes, OdsStringAttributes2Str, odsAccordionBaseAttributes } from "@ovhcloud/ods-testing";
 
 import { OdsThemeColorIntent } from '@ovhcloud/ods-theming';
+import { DEFAULT_ATTRIBUTE } from "./constants/default-attributes";
+import { OdsAccordionAttributes } from "./interfaces/attributes";
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/osds-accordion.e2e.screenshot.ts:5]: type

File: packages-new/components/accordion/src/components/osds-accordion/osds-accordion.e2e.screenshot.ts [hunk 17]
```diff
@@ -1,16 +1,18 @@
...
+import { OdsComponentAttributes2StringAttributes } from "@ovhcloud/ods-common-core";
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/osds-accordion.e2e.screenshot.ts:6]: please use `odsComponentAttributes2StringAttributes` from common-testing instead

File: packages-new/components/accordion/src/components/osds-accordion/osds-accordion.e2e.screenshot.ts [hunk 18]
```diff
@@ -1,16 +1,18 @@
...
+import { OdsStringAttributes2Str } from "@ovhcloud/ods-common-testing";
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/osds-accordion.e2e.screenshot.ts:7]: please use `odsStringAttributes2Str` instead
(sadly the bare copy of the libs have also kept things that will be removed soon)

File: packages-new/components/accordion/src/components/osds-accordion/osds-accordion.e2e.ts [hunk 19]
```diff
@@ -1,14 +1,15 @@
 import { E2EElement, E2EPage, newE2EPage } from "@stencil/core/testing";
-import { OdsAccordionAttributes, OdsComponentAttributes2StringAttributes, odsAccordionDefaultAttributes } from "@ovhcloud/ods-core";
-import { OdsCreateAttributes, OdsStringAttributes2Str, odsAccordionBaseAttributes } from "@ovhcloud/ods-testing";
+import { DEFAULT_ATTRIBUTE } from "./constants/default-attributes";
+import { OdsAccordionAttributes } from "./interfaces/attributes";
+import { OdsComponentAttributes2StringAttributes } from '@ovhcloud/ods-common-core';
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/osds-accordion.e2e.ts:4]: please use `odsComponentAttributes2StringAttributes` from common-testing instead

File: packages-new/components/accordion/src/components/osds-accordion/osds-accordion.e2e.ts [hunk 20]
```diff
@@ -1,14 +1,15 @@
...
+import { OdsStringAttributes2Str } from "@ovhcloud/ods-common-testing";
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/osds-accordion.e2e.ts:5]: please use `odsStringAttributes2Str` instead

File: packages-new/components/accordion/src/components/osds-accordion/osds-accordion.spec.ts [hunk 21]
```diff
@@ -1,12 +1,15 @@
-jest.mock('@ovhcloud/ods-core/src/components/accordion/ods-accordion-controller'); // keep jest.mock before any import
+jest.mock('./core/controller'); // keep jest.mock before any import
 
-import { OdsAccordionAttributes, OdsAccordionController, OdsAccordionSizeList, OdsComponentAttributes2StringAttributes, odsAccordionDefaultAttributes } from '@ovhcloud/ods-core';
-import { OdsCreateAttributes, OdsStringAttributes2Str, OdsUnitTestAttributeType, odsAccordionBaseAttributes, odsUnitTestAttribute } from '@ovhcloud/ods-testing';
-import { SpecPage, newSpecPage } from '@stencil/core/testing';
-
-import { OdsThemeColorIntentList } from '@ovhcloud/ods-theming';
+import { newSpecPage } from '@stencil/core/testing';
+import type { SpecPage } from '@stencil/core/testing';
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/osds-accordion.spec.ts:4]: type on top

File: packages-new/components/accordion/src/components/osds-accordion/osds-accordion.tsx [hunk 22]
```diff
@@ -1,49 +1,41 @@
...
-  OdsAccordion,
-  OdsAccordionController,
-  OdsAccordionEvents,
-  OdsAccordionGroup,
-  OdsAccordionMethods,
-  OdsAccordionSize,
-  odsAccordionDefaultAttributes
-} from '@ovhcloud/ods-core';
-import { OdsStencilEvents, OdsStencilMethods } from '@ovhcloud/ods-stencil/libraries/stencil-core';
 import { HTMLStencilElement } from '@stencil/core/internal';
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/osds-accordion.tsx:2]: type

File: packages-new/components/accordion/src/components/osds-accordion/osds-accordion.tsx [hunk 23]
```diff
@@ -1,49 +1,41 @@
...
 import { OdsThemeColorIntent } from '@ovhcloud/ods-theming';
 import { ODS_ICON_NAME, ODS_ICON_SIZE } from '@ovhcloud/ods-component-icon';
+import type { OdsAccordionAttributes } from './interfaces/attributes';
+import { DEFAULT_ATTRIBUTE } from './constants/default-attributes';
+import type { ODS_ACCORDION_SIZE } from './constants/accordion-size';
+import { OdsAccordionController } from './core/controller';
+import { OsdsAccordionGroup } from '../osds-accordion-group/osds-accordion-group';
```
Comment (dpellier) [inline packages-new/components/accordion/src/components/osds-accordion/osds-accordion.tsx:9]: type

File: packages/tools/storybook/stories/components/accordion-group/accordion-group.web-component.stories.page.mdx [hunk 24]
```diff
@@ -1,7 +1,7 @@
 import { Canvas, Description, Meta, DocsContainer } from '@storybook/addon-docs';
 
-import Usage from '@ovhcloud/ods-stencil-accordion/docs/osds-accordion-group/usage.mdx';
-import APITable from '@ovhcloud/ods-stencil-accordion/dist/docs/components/osds-accordion-group/readme.md';
+import Usage from '@ovhcloud/ods-component-accordion/documentation/osds-accordion-group/usage.mdx';
```
Comment (dpellier) [inline packages/tools/storybook/stories/components/accordion-group/accordion-group.web-component.stories.page.mdx:3]: still generated in docs for now, will be moved later

File: packages/tools/storybook/stories/components/accordion/accordion.web-component.stories.page.mdx [hunk 25]
```diff
@@ -1,7 +1,7 @@
 import { Canvas, Description, Meta, DocsContainer } from '@storybook/addon-docs';
 
-import Usage from '@ovhcloud/ods-stencil-accordion/docs/osds-accordion/usage.mdx';
-import APITable from '@ovhcloud/ods-stencil-accordion/dist/docs/components/osds-accordion/readme.md';
+import Usage from '@ovhcloud/ods-component-accordion/documentation/osds-accordion/usage.mdx';
```
Comment (dpellier) [inline packages/tools/storybook/stories/components/accordion/accordion.web-component.stories.page.mdx:3]: still generated in docs for now, will be moved later

Comment (aesteves60) [reply packages/tools/storybook/stories/components/accordion/accordion.web-component.stories.page.mdx:3]: i think it's not auto generate

**Comment by ovh-cds** [comment]
CDS Report build#1137.0 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#1137.1 ✘  
* 
  * build ✘

**Comment by ovh-cds** [comment]
CDS Report build#1157.0 ✘  
* 
  * build ✘

**Pull request closed without merging**